File: easy/3712_sum_of_elements_with_frequency_divisible_by_k/solution.cpp
#include <numeric>
#include <unordered_map>
#include <vector>

class Solution {
public:
  int sumDivisibleByK(std::vector<int> &nums, int k) {
    std::unordered_map<int, int> numsCounter;
    for (const int &num : nums)
      ++numsCounter[num];

    int numsSum = 0;

    for (const auto &[num, count] : numsCounter) {
      if (count % k == 0)
        numsSum += num * count;
    }

    return numsSum;
  }
};

File: easy/3712_sum_of_elements_with_frequency_divisible_by_k/Solution.java
class Solution {
  public int sumDivisibleByK(int[] nums, int k) {
    final int[] numsCounter = new int[101];
    for (final int num : nums) {
      ++numsCounter[num];
    }

    int numsSum = 0;

    for (int i = 0; i < 101; ++i) {
      if (numsCounter[i] % k == 0) {
        numsSum += i * numsCounter[i];
      }
    }

    return numsSum;
  }
}
